homeassistant/components/sensor/modbus.py

```diff
@@ -70,7 +70,7 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         structure = '>i'
         if register.get(CONF_DATA_TYPE) != DATA_TYPE_CUSTOM:
             try:
-                structure = '>{:c}'.format(data_types[
+                structure = '>{}'.format(data_types[
                     register.get(CONF_DATA_TYPE)][register.get(CONF_COUNT)])
```

Comment:
This whole line doesn't make any sense. You're using the count as the key to look up off of the data_types elements above, but they only have entries for 1, 2, and 4.

Reply:
Is correct.... I've never seen a modbus device that split a number across 3 o 5 registers.  1, 2 and 4 cover 99% of cases.  Remaining 1% is particular and can be coverd using a custom unpack string.

This pr is necessary because with last commit on modbus sensor the component stop working if using not custom types.

Reply:
Oof, OK, that makes sense. Count got me thinking of the number of values, but you're absolutely right, this is for a single sensor value.

Reply:
👍 Thanks!